Documentation/cheatsheet.rst

```diff
@@ -2,157 +2,396 @@
 ::
 
-	# Enable debugging output on the cilium monitor for this endpoint
+    cilium endpoint list
 
-	cilium endpoint config <id> Debug=true
+Get detailed view of endpoint properties and state
+::
 
-Tracing
-=======
+    cilium endpoint get <id>
 
+Show recent endpoint specific log entries
 ::
 
-	# Check policy enforcement between two labels on port 80
+    cilium endpoint log <id>
 
-	cilium policy trace -s <app.from> -d <app.to> --dport 80
+Enable debugging output on the cilium monitor for this endpoint
+::
+
+    cilium endpoint config <id> Debug=true
 
+
+Loadbalancing
+-------------
+
+Get list of loadbalancer services
 ::
 
-        # Check policy enforcement between two identities
+    cilium service list
+
+
+Or you can get the loadbalancer information using bpf list
+:::
 
-        cilium policy trace --src-identity <from-id> --dst-identity <to-id>
+    cilium bpf lb list
 
+
+Add a new loadbalancer
 ::
 
-        # Check policy enforcement between two pods
+    cilium service update --frontend 127.0.0.1:80 \
+        --backends 127.0.0.2:90,127.0.0.3:90 \
+        --id 20 \
+        --rev 2
 
-        cilium policy trace --src-k8s-pod <namespace>:<pod.from> --dst-k8s-pod <namespace>:<pod.to>
+BPF
+---
 
-Loadbalancing
-=============
+List node tunneling mapping information
+::
+
+    cilium bpf tunnel list
 
+Checking logs for verifier issue
+::
+
+    journalctl -u cilium | grep -B20 -F10 Verifier
 
+List connection tracking entries:
 ::
 
-        # Get list of loadbalancer services
+    sudo cilium bpf ct list global
 
-        cilium service list
+Flush connection tracking entries:
+::
 
-BPF
-===
+    sudo cilium bpf ct flush
 
+List proxy configuration:
 ::
 
-        # List node tunneling mapping information
+    sudo cilium bpf proxy list
+
+
+Kubernetes examples:
+=====================
+
+If you running Cilium on top of Kubernetes you may also want a way to list all
+cilium endpoints or policies from a single Kubectl commands. Cilium by using
+`Kubernetes Resource Definitions
+<https://kubernetes.io/docs/concepts/api-extension/custom-resources/>`_ provides
+all this information to the user:
+
+
+Policies
+---------
+
+In Kubernetes you can use two kind of policies, Kubernetes Network Policies or
+cilium network polcies. Both can be retrieved from the ``kubectl`` command:
 
-        cilium bpf tunnel list
+.. code-block:: bash
+   :name: Kubernetes Network Policies
+   :caption: Kubernetes Network Policies
+
+    kubectl get netpol
+
+.. code-block:: bash
+   :name: Kubernetes Cilium Policies
+   :caption: Kubernetes Cilium Policies
+
+    $ kubectl get cnp
+    NAME      AGE
+    rule1     3m
+    $ kubectl get cnp rule1
+    NAME      AGE
+    rule1     3m
+    $ kubectl get cnp rule1 -o json
+
+
+Endpoints
+----------
+
+To retrieve a list of all endpoints managed by cilium, a ``Cilum Endpoint``
+command can be used.
 
 ::
 
-        # Checking logs for verifier issue
+    $ kubectl get cep
```

Comment:
maybe add `--all-namespaces` as a subtle hint that these CRDs are namespace scoped.

Reply:
for some reason I thought the CNPs were namespace agnostic. They are, in fact, namespaced so ignore me.